packages/extension/src/view/devtools/components/cookies/cookiesListing/cookieTableContainer/editableComponents/editableTextInput.tsx

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies.
+ */
+import React, { useCallback, useState, useRef, useEffect, memo } from 'react';
+
+/**
+ * Internal dependencies.
+ */
+import { useCookieStore } from '../../../../../stateProviders/syncCookieStore';
+import type { InfoType } from '@ps-analysis-tool/design-system';
+import { noop } from '@ps-analysis-tool/common';
+
+interface EditableTextInputProps {
+  info: InfoType;
+  keyToChange: string;
+  cookieKey?: string | null;
+  rowHighlighter?: (value: boolean, cookieKey: string) => void;
+}
+
+const EditableTextInput = ({
+  info,
+  keyToChange,
+  cookieKey,
+  rowHighlighter = noop,
+}: EditableTextInputProps) => {
+  const [editing, setEditing] = useState(false);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const [localValue, setLocalValue] = useState<string>(info as string);
+  const { modifyCookie } = useCookieStore(({ actions }) => ({
+    modifyCookie: actions.modifyCookie,
+  }));
+
+  const handleDoubleClick = useCallback(
+    (event: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
+      if (event.detail === 2) {
+        setEditing(true);
+      }
+    },
+    []
+  );
+
+  useEffect(() => {
+    if (editing) {
+      inputRef.current?.select();
+      inputRef.current?.setSelectionRange(0, -1);
+    }
+  }, [editing]);
+
+  const handleClickOutside = useCallback(
+    async (event: MouseEvent) => {
+      if (
+        inputRef.current &&
+        !inputRef.current.contains(event.target as Node)
+      ) {
+        setEditing(false);
+        if (localValue !== info && cookieKey) {
+          const result = await modifyCookie(
```

Comment:
```suggestion
          const isUpdateDone = await modifyCookie(
```

Please check for other editable components.

Reply:
Fixed in latest commit.